test/es-module/test-esm-loader-resolve-type.mjs

```diff
@@ -0,0 +1,42 @@
+// Flags: --loader ./test/fixtures/es-module-loaders/hook-resolve-type.mjs
+import { allowGlobals } from '../common/index.mjs';
+import * as fixtures from '../common/fixtures.mjs';
+import { strict as assert } from 'assert';
+import * as fs from 'fs';
+import { fileURLToPath, URL } from 'url';
+
+allowGlobals(global.getModuleTypeStats);
+
+const basePath =
+  new URL('./node_modules/', import.meta.url);
+
+const rel = (file) => new URL(file, basePath);
+const createDir = (path) => {
+  if (!fs.existsSync(path)) {
+    fs.mkdirSync(path);
+  }
+};
+
+const moduleName = 'module-counter-by-type';
+
+const moduleDir = rel(`${moduleName}`);
+createDir(basePath);
+createDir(moduleDir);
+fs.cpSync(
+  fixtures.path('es-modules', moduleName),
+  moduleDir,
+  { recursive: true }
+);
+
+const { importedESM: importedESMBefore,
+        importedCJS: importedCJSBefore } = global.getModuleTypeStats();
+
+import(`${moduleName}`).finally(() => {
+  fs.rmSync(fileURLToPath(basePath), { recursive: true, force: true });
```

Comment:
That is not necessary, all `fs` functions accept URLs as first parameter.
```suggestion
  fs.rmSync(basePath, { recursive: true, force: true });
```

Reply:
https://github.com/nodejs/node/pull/41132 has landed.

Reply:
dependency resolved with commit: https://github.com/nodejs/node/pull/40980/commits/2d2c4754322117dd5ccf362206e45be42c0edfb8